portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson; 
+import com.google.gson.GsonBuilder;  
+import java.util.HashMap;
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    // logInOutUrl is either the login or logout url depending on user status
+    String logInOutUrl;
+    boolean isLoggedIn;
+
+    // generate login/logout url & login status as 0/1
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      logInOutUrl = userService.createLogoutURL("/");
+    //   ;StartPollingForCompletion();return false;">
+      isLoggedIn = true;
+    } else {
+      logInOutUrl = userService.createLoginURL("/");
+      isLoggedIn = false;
+    }
+
+    Gson gsonBuilder = new GsonBuilder().create();
+    HashMap loginDataMap = new HashMap();
+
+    // add status and url to hashmap
```

Comment:
This comment isn't necessary, as it's clear that it's adding the URL and whether the user's logged in (rather than status) to the JSON. If it's doing something unusual or unintuitive, then here would be a good reason to explain why

Reply:
Thanks